packages/backend/src/accounting/psql/service.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Asset } from '../../asset/model'
+import { AssetService } from '../../asset/service'
+import { BaseService } from '../../shared/baseService'
+import { CreateAccountError, TransferError } from '../errors'
+import {
+  AccountingService,
+  Deposit,
+  LiquidityAccount,
+  LiquidityAccountType,
+  Transaction,
+  TransferOptions,
+  Withdrawal
+} from '../service'
+import {
+  LedgerAccountType,
+  mapLiquidityAccountTypeToLedgerAccountType
+} from './ledger-account/model'
+import { LedgerAccountService } from './ledger-account/service'
+
+export interface ServiceDependencies extends BaseService {
+  ledgerAccountService: LedgerAccountService
+  withdrawalThrottleDelay?: number
+}
+
+export function createAccountingService(
+  deps_: ServiceDependencies
+): AccountingService {
+  const deps = {
+    ...deps_,
+    logger: deps_.logger.child({ service: 'PsqlAccountingService' })
+  }
+  return {
+    createLiquidityAccount: (options, accTypeCode) =>
+      createLiquidityAccount(deps, options, accTypeCode),
+    createSettlementAccount: (ledger) => createSettlementAccount(deps, ledger),
+    getBalance: (id) => getAccountBalance(deps, id),
+    getTotalSent: (id) => getAccountTotalSent(deps, id),
+    getAccountsTotalSent: (ids) => getAccountsTotalSent(deps, ids),
+    getTotalReceived: (id) => getAccountTotalReceived(deps, id),
+    getAccountsTotalReceived: (ids) => getAccountsTotalReceived(deps, ids),
+    getSettlementBalance: (ledger) => getSettlementBalance(deps, ledger),
+    createTransfer: (options) => createTransfer(deps, options),
+    createDeposit: (transfer) => createAccountDeposit(deps, transfer),
+    createWithdrawal: (transfer) => createAccountWithdrawal(deps, transfer),
+    postWithdrawal: (options) => postAccountWithdrawal(deps, options),
+    voidWithdrawal: (options) => voidAccountWithdrawal(deps, options)
+  }
+}
+
+export async function createLiquidityAccount(
+  deps: ServiceDependencies,
+  account: LiquidityAccount,
+  accountType: LiquidityAccountType
+): Promise<LiquidityAccount> {
+  await deps.ledgerAccountService.create({
+    accountRef: account.id,
+    ledger: account.asset.ledger,
+    type: mapLiquidityAccountTypeToLedgerAccountType[accountType]
+  })
+
+  return account
+}
+
+export async function createSettlementAccount(
+  deps: ServiceDependencies,
+  ledger: number
+): Promise<void> {
+  const asset = await Asset.query(deps.knex).findOne({ ledger })
```

Comment:
I _dont_ like using the `Asset` model here directly but if I were to pull in `assetService`, it would be a circular dependency 😞 

This isn't the ideal solution, but at least allows us to not have to change any of the AccountingService function signatures & allows us to support both the psql tables and tigerbeetle service

Reply:
Yup, good call